hs/WebParsing/TimeTableParser.hs

```diff
@@ -17,15 +18,111 @@ import WebParsing.TimeConverter
 import Control.Monad.IO.Class (liftIO, MonadIO)
 import Control.Monad.Trans.Reader (ReaderT)
 import Config (databasePath, timetableUrl)
+import Database.Tables (Department(Department))
 
 parseTT :: IO ()
 parseTT = do
     rsp <- simpleHTTP (getRequest timetableUrl)
     body <- getResponseBody rsp
+
     let depts = getDeptList $ parseTags body
+    let deptCode = getDeptCode $ parseTags body
+    let deptName = getDeptName $ parseTags body
 
     runSqlite databasePath $
         mapM_ getDeptTimetable depts
+    runSqlite databasePath $ 
+        mapM_ insertDepartment $ init $ constructAssociationList deptName deptCode
+
+-- | String left bracket to split tagText into two strings. One containing the 
+-- department name and the other the department codes. 
+bracketLeftString :: String
+bracketLeftString = "["
+
+-- | Char left bracket used to call removeUnwantedChar. Removes all char prior to 
+-- and including itself from given String. 
+bracketLeft :: Char
+bracketLeft = '['
+
+-- | Char right bracket to remove from the end of the last words in tagTexts.
+bracketRight :: Char
+bracketRight = ']'
+
+-- | List of unwanted words to remove when getting department codes.
+notDeptCodes :: [String]
+notDeptCodes = ["courses",
+                "courses]", 
+                "includes",
+                "and",
+                "-",
+                "[",
+                "]",
+                ""]
+
+-- | Parses the timetable main page, retrieves the tag text and gets the department
+-- codes in a list of lists of T.Text.
+getDeptCode :: [Tag String] -> [[T.Text]]
```

Comment:
Hmm. Is there any way that we can better represent the data? Why do some departments have more than one code?